### ISSUE TITLE
`@types/chrome` needs to be installed manually when using pnpm

### ISSUE THREAD

**Issue by 2wheeh** [wxt-dev/wxt#1116]
### Describe the bug

when using npm, TS infers browser namespace successfully:

<img width="695" alt="image" src="https://github.com/user-attachments/assets/3257189f-695a-40ef-bcf1-a8db385dcaa4">

but when using pnpm, it fails without installing `@types/chrome` manually:

<img width="558" alt="image" src="https://github.com/user-attachments/assets/401dd741-c21e-4e56-a42b-b36f38ab9352">

we may need to improve docs or fix: this might be related to how pnpm handles devDependencies differently from npm.

### Reproduction

just start fresh projects with pnpm following installation docs.
```shell
pnpm dlx wxt@latest init
```

### Steps to reproduce

after installing, put your cursor on `browser.runtime.*` to see the inferred types. 
for example, `browser.runtime.id` in `entrypoints/background.ts`.

### System Info

```shell
System:
    OS: macOS 15.0.1
    CPU: (11) arm64 Apple M3 Pro
    Memory: 2.53 GB / 36.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.18.0 - ~/.nvm/versions/node/v20.18.0/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.18.0/bin/npm
    pnpm: 9.9.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 130.0.6723.70
    Safari: 18.0.1
  npmPackages:
    wxt: ^0.19.11 => 0.19.13
```

### Used Package Manager

pnpm

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/wxt-dev/wxt/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://wxt.dev/guide/installation.html).
- [X] Check that there isn't [already an issue](https://github.com/wxt-dev/wxt/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/wxt-dev/wxt/discussions) or join our [Discord Chat Server](https://discord.gg/ZFsZqGery9).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by 1natsu172** [comment]
Is the `extensionApi` in config set to `'chrome'`? Also try restarting ts-server on your editor. ref: https://wxt.dev/guide/essentials/extension-apis.html#disabling-the-polyfill

Although you are still experiencing differences in behavior due to the package manager?

**Comment by 2wheeh** [comment]
@1natsu172 

> Is the extensionApi in config set to 'chrome'? Also try restarting ts-server on your editor.

yes, it is set to 'chrome' as configured after scaffolding the project via `pnpm dlx wxt@latest init` currently.

> Although you are still experiencing differences in behavior due to the package manager?

yes, there seems to be an issue where `browser.runtime` fails to load properly.

## without `@types/chrome` on the app devDependency
<img width="810" alt="image" src="https://github.com/user-attachments/assets/60e4d489-3f0f-4a74-856c-2e9657a132ca">

<img width="778" alt="image" src="https://github.com/user-attachments/assets/7aeba9eb-6fe3-4d8b-9eb3-10caee527ab7">

## with `@types/chrome` on the app devDependency

<img width="886" alt="image" src="https://github.com/user-attachments/assets/419f586e-4e54-4ac4-8b3b-249100024f4f">

<img width="888" alt="image" src="https://github.com/user-attachments/assets/6031c3d7-83df-402c-a1da-74e685f9c3b8">

plus, even in npm (or with `@types/chrome` in pnpm), the type inference fails for the snippet below from the docs while it works correctly with `chrome` namespace:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/c1aab944-194f-47d0-86aa-7efbe97df966">

<img width="790" alt="image" src="https://github.com/user-attachments/assets/e83f3e6e-91b4-4c83-b31a-b9a39cf4d3b6">

<img width="644" alt="image" src="https://github.com/user-attachments/assets/404e51ea-0f38-428f-b714-9f8d19a6341e">

if this isn't a bug I may be missing something important here - would you have any suggestions?

**Comment by 2wheeh** [comment]
as per https://github.com/wxt-dev/wxt/issues/784#issuecomment-2248989854, it seems to need to install `@types/chrome` when using pnpm.

when `browser` is a value it infers well but when it is just a type, it shows ts error.

**Comment by 1natsu172** [comment]
@2wheeh Alright, there are two problems. I have knew that too. I understood it too.

First, for some reason in the `pnpm` environment, the internal dependency `@types/chrome` can't be resolved and the `typeof chrome` is failing. The workaround to this problem is to include `@types/chrome` in the devDeps on the app side, as you already understood.

Here's the code inside.
https://github.com/wxt-dev/wxt/blob/68183b648e309641e415f5856d233bea6f5f4439/packages/wxt/src/browser/chrome.ts#L12

---

The second thing is that the interface _as a type_ cannot be inferred from the `browser` as you say. This one is not a package-manager dependent problem. I think it's an implementation glitch in WXT.

The workaround over here is below, and either approach should be used.

```typescript
import { Runtime } from "wxt/browser";

// This types from webextension-polyfill
function handler1(params: Runtime.MessageSender) {}

// This types from @types/chrome(need install `@types/chrome`)
function handler2(params: chrome.runtime.MessageSender) {}
```

**Comment by 2wheeh** [comment]
@1natsu172 thank you for the very clear summary 👍🏻

**Comment by aklinker1** [comment]
Alright, so `@types/chrome` can't be resolved because when it's a subdependency, it's located at `node_modules/.pnpm/@types/chrome`. However, the triple-slash reference WXT adds (`/// <reference types="@types/chrome" />`) only looks for `node_modules/@types/chrome`, it doesn't look inside `.pnpm` directory, so it doesn't add the types correctly.

So the solution is to use `import.resolve` to find the true file path of `@types/chrome`, and use that in the reference instead of just `"@types/chrome"` here:

https://github.com/wxt-dev/wxt/blob/54baf21490c996cce441a5183e04a8c89725941c/packages/wxt/src/core/generate-wxt-dir.ts#L42

**Comment by aklinker1** [comment]
I'll try and get to this soon, but if someone else wants to give it a go, see [my preview comment](https://github.com/wxt-dev/wxt/issues/1116#issuecomment-2498821784).

**Comment by 1natsu172** [comment]
Another issue is more complex, so I have split it into a separate issue. #1213

**Comment by 2wheeh** [comment]
> So the solution is to use import.resolve to find the true file path of @types/chrome, and use that in the reference instead of just "@types/chrome" here:

So `getMainDeclarationEntry` needs to be updated to resolve correct module path.
```ts
 if ('module' in ref) {
      return lines.push(`/// <reference types="${ref.module}" />`);
 }
``` 

but the thing is it seems not possible to use `import.meta.*` here...

`ERROR  Cannot use 'import.meta' outside a module` :/

something around `jiti` or `c12` might be related?

**Comment by aklinker1** [comment]
@2wheeh If you base off [`0.20.0-breaking-changes`](https://github.com/wxt-dev/wxt/pull/1085), I've already [merged the fix this issue](https://github.com/wxt-dev/wxt/pull/1182). If you open a PR into that branch, it will go out in the next major version.

But we already have a method for finding the path to a NPM package in a CJS/ESM environment without `import.meta`: 

https://github.com/wxt-dev/wxt/blob/241c907c65c23dfb786a8faf35ba5a435a4d2dbe/packages/wxt/src/core/resolve-config.ts#L423-L438

You could refactor this function to work with any NPM package. Then we could get this fix out before 0.20

**Comment by aklinker1** [comment]
Released in [v0.20.0](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.20.0). Since we no longer use global namespaces for types, this is fixed!
